node/pkg/terra/watcher.go

```diff
@@ -156,6 +167,63 @@ func (e *Watcher) Run(ctx context.Context) error {
 		}
 	}()
 
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDTerra {
+					panic("invalid chain ID")
+				}
+
+				tx := eth_common.BytesToHash(r.TxHash)
```

Comment:
It's a bit weird to use a go-ethereum Hash type here - any specific reason we do that?

Reply:
prior art. no specific reason, we use this same thing in eth and in StringToHash at the bottom of the file

Reply:
StringToHash uses it because it explicitly needs a common.Hash for MessageObservation. It's just a needlessly complicated way of converting plain bytes to hex otherwise.